### PR TITLE
PXY-609-port-in-use-doesnt-exit

### PR DESCRIPTION
**Pull request by manhcuongbk56**
# What does this pr do? 

* change the way GatewayThread run, so it will respect time left on shutting down period
* add Gateway start fail exception so we can show reason why proxy start fail to user

# Note
Verified by manual test.

### CODE REVIEW

**Review by framiere — APPROVED** [review]

File: gateway-core/src/main/java/io/conduktor/gateway/Bootstrap.java [hunk 1]
```diff
@@ -120,7 +121,11 @@ private static void startGateway(Injector injector) {
             app = injector.getInstance(GatewayExecutor.class);
             app.start();
         } catch (Exception ex) {
-            log.error("Error happen when start app", ex);
+            if (ex instanceof GatewayStartFailException startFailException) {
+                log.error(startFailException.getReason() + ". Gateway can not start!!!");
+            } else {
+                log.error("Error happen when start app", ex);
```
Comment (thomaskwscott) [inline gateway-core/src/main/java/io/conduktor/gateway/Bootstrap.java:127]:
```suggestion
                log.error("An error occurred when starting Gateway", ex);
```

File: gateway-core/src/main/java/io/conduktor/gateway/GatewayExecutor.java [hunk 2]
```diff
@@ -69,7 +73,7 @@ public void close() {
                 brokerManager.close();
             }
         } catch (Exception e) {
-            log.error("Error happen when close metric registry provider");
+            log.error("Error happen when shutting down gateway", e);
```
Comment (thomaskwscott) [inline gateway-core/src/main/java/io/conduktor/gateway/GatewayExecutor.java:76]:
```suggestion
            log.error("An error occured when shutting down Gateway", e);
```

File: gateway-core/src/main/java/io/conduktor/gateway/network/GatewayBrokers.java [hunk 3]
```diff
@@ -96,9 +97,12 @@ public void acquirePort(int port, GatewayChannelInitializer channelInitializer)
...
             gauge.measure();
             log.info("Gateway channel bound: {}", channel.localAddress());
-        } catch (Exception e) {
-            log.error("Error when init new port of Gateway: [port={}]", port, e);
-            throw new RuntimeException(e);
+        } catch (Exception exception) {
+            if (exception instanceof BindException) {
+                throw new GatewayStartFailException("port " + port + " is in used!!!");
+            }
+            log.error("Unknown error when init new port of Gateway: [port={}]", port, exception);
```
Comment (thomaskwscott) [inline gateway-core/src/main/java/io/conduktor/gateway/network/GatewayBrokers.java:104]:
```suggestion
            log.error("An unknown error occurred when initialising a new port for Gateway: [port={}]", port, exception);
```

File: gateway-core/src/main/java/io/conduktor/gateway/thread/GatewayThread.java [hunk 4]
```diff
@@ -271,7 +295,7 @@ private void rebuild(ByteBuf buf, GatewayChannel gatewayChannel) {
                         return null;
                     });
         } catch (Exception exception) {
-            logErrorIfRequired(String.format("Caught Exception when error happened when sending request to kafka cluster at: {}", requestHeader),
+            logErrorIfRequired(String.format("Caught Exception when error happened when sending request to kafka cluster at: %s", requestHeader),
```
Comment (thomaskwscott) [inline gateway-core/src/main/java/io/conduktor/gateway/thread/GatewayThread.java:298]:
```suggestion
            logErrorIfRequired(String.format("An exception was thrown when during request send to Kafka cluster at: %s", requestHeader),
```

**Review by thomaskwscott — APPROVED** [review]
Grammar nits but otherwise LGTM

**Pull request merged**